tests/gas_cost_test.go

```diff
@@ -265,13 +265,20 @@ func testGasCosts_Allegro(t *testing.T, singleProposer bool) {
 
 				expectedCost, err := core.IntrinsicGas(tx.Data(), tx.AccessList(), tx.SetCodeAuthorizations(), tx.To() == nil, true, true, true)
 				require.NoError(t, err)
-				unused := tx.Gas() - expectedCost
-				if !singleProposer {
-					expectedCost += unused / 10
-				}
 
-				if floorDataGas > expectedCost {
+				if floorDataGas <= expectedCost {
+					if !singleProposer {
+						unused := tx.Gas() - expectedCost
+						expectedCost += unused / 10
+					}
+				} else {
 					expectedCost = floorDataGas
+
+					// Even if the floor data gas has to be paid, we still charge 10% of unused gas
+					unused := tx.Gas() - expectedCost
+					if !singleProposer {
+						expectedCost += unused / 10
+					}
```

Comment:
Would this be the same?
```suggestion
				if floorDataGas > expectedCost {
					expectedCost = floorDataGas
					expectedSmallerThanFloor++
				}
				if !singleProposer {
					unused := tx.Gas() - expectedCost
					expectedCost += unused / 10
				}
```